pybamm/solvers/casadi_solver.py

```diff
@@ -104,7 +104,7 @@ def __init__(
 
         pybamm.citations.register("Andersson2019")
 
-    def _integrate(self, model, t_eval, inputs_dict=None):
+    def _integrate(self, model, t_eval, inputs_dict=None, ninputs=1, nproc=1):
```

Comment:
Why have a separate `ninputs` here instead of reading the length of `inputs_dict`?
Also if `inputs_dict` can now be a list or a dict, we should rename it to avoid confusion (maybe even just to `input_dicts`?)